### ISSUE TITLE
Performance between 2 components

### ISSUE THREAD

**Issue by henricavalcante** [facebook/react#3616]
I'm trying to proof that React is faster than Angular, but I had a problem with my code.

I need to generate a list with 1000 lines and use a input to change the value of specific field inside then, in angular I can do a fast binding, however in react I can't.

This is my code in angular:

```html
<!DOCTYPE html>
<html>
	<head>
		<script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.3.14/angular.min.js"></script>
	</head>  
	<body ng-app="myApp" ng-controller="MainContrl">

		<input type="text" ng-model="bind" />
		<ul>
			<li ng-repeat="r in range">
				{{r}} - {{bind}}
			</li>
		</ul>
	<script>
		var app = angular.module('myApp', []);
			app.controller('MainContrl', function($scope) {
				var d = new Date();
				var range = [];

				for(var i = 0; i <= 1000; i++) {
				  range.push(i + ' - ' + d.getTime());
				}

				$scope.range = range;
		});
	</script>
  </body>
</html>
```

And this is my code in react:
``` html
<!DOCTYPE html>
<html>
	<head>
		<script src="http://fb.me/react-0.13.1.js"></script>
		<script src="http://fb.me/JSXTransformer-0.13.1.js"></script>
	</head>  
	<body>
		<div id="content"></div>
		
		<script type="text/jsx">

			var Span = React.createClass({
			  getInitialState: function() {
			    return {
			      value: ''
			    };
			  },
			  componentDidMount: function() {
			    document.addEventListener('event-input',this.handleChange);
			  },
			  handleChange: function(e) {
			  	this.setState({msg: e.detail});
			  },
			  componentWillUnmount: function() {
			    document.removeEventListener('event-input');
			  },
				render: function () {
					return (
						<span key={this.props.chave}>{this.state.msg}</span>
					);
				}
			});

			var Hello = React.createClass({
		    handleChange: function (e) {
		      var event = new CustomEvent('event-input', { 'detail': e.target.value });
					document.dispatchEvent(event);
		    },
		    render: function() {
		    	var li = [];
		    	for (var i = 0; i <= 1000; i++)
      		{
      			li.push({
      				id: i,
      				time: (new Date()).getTime()
    				});
      		}
		      return (
		        <div>{(new Date()).getTime()}
		          <input type="text" onChange={this.handleChange} />
		          <ul>
							{li.map(function(result) {
			          return <li key={result.id}>{result.id} - {result.time} - <Span chave={'span'+result.id} /></li>;
			        })}
		          </ul>
		        </div>
		      );
		    }
		  });

		  React.render(<Hello />, document.getElementById('content'));
		</script>
	</body>
</html>
```

Can somebody help me to proof that React is faster than Angular?

**Comment by camspiers** [comment]
Something like this? http://jsfiddle.net/5jpr3o54/1/

**Comment by henricavalcante** [comment]
compare with angular: http://jsfiddle.net/vdm7aoaq/

I need the same performance with React to convince co-workers to change the technology.

**Comment by syranide** [comment]
@henricavalcante Neither is absolutely fastest, it doesn't work like that and it's not a meaningful metric. You can construct cases where React will win hands down and cases where Angular wins hands down. React, by and large, tend to be much more manageable as applications scale up, performance is usually predictable and hotspots can mostly be very easily addressed. Anecdotally this is not the case for Angular and friends, which tend to degrade as applications scale up and it becomes very hard to do something about it.

Regardless, first of all it's running the debug version of React which is significantly slower and this specific example is the worst-case for React, I don't see why you would ever want to update 1000+ items per keypress like you do here.

**Comment by henricavalcante** [comment]
Unfortunately, thanks for help.
